libr/util/file.c

```diff
@@ -1549,11 +1549,17 @@ R_API RList* r_file_glob(const char *_globbed_path, int maxdepth) {
 		if (last_slash) {
 			glob_ptr = last_slash + 1;
 			if (globbed_path[0] == '~') {
-				char *rpath = r_str_newlen (globbed_path + 2, last_slash - globbed_path - 1);
+				char *rpath = NULL;
+				if (last_slash - globbed_path - 1) {
```

Comment:
dont compute this twice. like in all the other cases. this is R_STR_NDUP imho. because we want to return null when the len is negative. but also we have the rstrget that performs a null check. so maybe this r_str_get should be also a macro in uppercase. this way we unify the logic of:

uppercase = macro, check for nulls, negative offsets accepted
lowercase = apis, dont accept nulls, negative offsets err